### PR TITLE
DAH-11 chore(backend): Setup Express

### PR DESCRIPTION
**Pull request by jaysabe**
## Pull Request
 Added Express.js to the backend folder and updated .gitignore file

### Description

Completed Express.js set up in server directory. Then added json. 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)\
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### Issue ticket number and link

[DAH-11](https://igit.atlassian.net/jira/software/projects/DAH/boards/4?selectedIssue=DAH-11)

### Checklist

- [x] Self-reviewed my code prior to submission.
- [ ] Tests have been added or updated (if applicable).
- [ ] Documentation has been updated (if applicable).
- [ ] Screenshots added (if applicable).
- [x] This PR is ready for review.

### Screenshots

[DAH-11]: https://igit.atlassian.net/browse/DAH-11?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by RickyC0626 — CHANGES_REQUESTED** [review]
**thought:** I think you meant to put `package.json` in `backend` directory?
**suggestion:** Remove `node_modules` and move `package.json` and `package-lock.json` to `backend` directory. Perhaps we can also add another `.gitignore` file in root directory in case.

File: backend/server/server.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+"use strict";
+
+//Import requirement modules
+const express = require("express");
+const app = express();
+const port = 3000;
+
+test = app.get("/", (req, res) => {
```
Comment (RickyC0626) [inline backend/server/server.js]: **suggestion:** Remove `test =` assignment, `app.get` will work without it.

File: backend/app.js [hunk 2]
```diff
@@ -0,0 +1 @@
+"use strict";
```
Comment (RickyC0626) [inline backend/app.js:1]: **thought:** This file is empty, maybe `server.js` could be moved here instead?

File: backend/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "dahlia-1",
+  "version": "1.0.0",
+  "description": "parking app",
+  "main": "index.js",
+  "scripts": {
+    "test": "test_run"
+  },
```
Comment (RickyC0626) [inline backend/package.json]: **suggestion:** A `start` or `dev` script to run the app would be nice.

`dev` script could use something like [`nodemon`](https://nodemon.io) for hot reload.

`start` script could just run `node app.js`.

**Review by RickyC0626 — CHANGES_REQUESTED** [review]

File: backend/app.js (hunk 2 above)
Comment (jaysabe) [reply backend/app.js:1]: Are you saying to bring server.js out to just backend/ ?

Comment (RickyC0626) [reply backend/app.js:1]: Yeah, the contents of `server.js` can go here.

File: backend/package.json (hunk 3 above)
Comment (jaysabe) [reply backend/package.json]: added both for app.js

File: backend/.gitignore [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# IDE and editor files
+.idea/
+.vscode/
+
+# Test coverage reports
+coverage/
+
+# Miscellaneous
+*.log
+*.env
```
Comment (RickyC0626) [inline backend/.gitignore]: **suggestion:** 
```diff
*.env
+ !.env.example
```
This will be useful later for setting up env vars

File: backend/.gitignore [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# dependencies
+node_modules
+# Development logs
+logs/
+
+# dotenv environment variables file
+.env
```
Comment (RickyC0626) [inline backend/.gitignore]: **thought:** Oh, we have a duplicate. Feel free to move `*.env` up here.

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+# IDE and editor files
+.idea/
+.vscode/
+
+# Test coverage reports
+coverage/
+
+# Miscellaneous
+*.log
+*.env
```
Comment (RickyC0626) [inline .gitignore]: **suggestion:** Same changes as `backend/.gitignore`

**Review by RickyC0626 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 6 above)
Comment (jaysabe) [reply .gitignore]: Are you saying change in frontend .gitignore to match ?

**Review by RickyC0626 — APPROVED** [review]

**Pull request merged**